Error while checking build/O2/fullCI for e09eedbbb4e1be7a0cbdf573ca61e531800c5646 at 2023-11-30 16:45:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/QualityControl-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile digiTRD.log
digiTRD.log-[42159:internal-dpl-ccdb-backend]: [15:44:59][FATAL] Unable to find object CTP/Calib/OrbitReset/1550600800000
digiTRD.log:[42159:internal-dpl-ccdb-backend]: [15:44:59][ERROR] Exception caught: Fatal error 
[42159:internal-dpl-ccdb-backend]: [15:44:59][ERROR] Failed to open file alien:///alice/data/CCDB/CTP/Calib/OrbitReset/01/35179/b07c53e0-b4c0-11ec-b66d-90ce809b250c?filetype=raw
[42159:internal-dpl-ccdb-backend]: [15:44:59][FATAL] Unable to find object CTP/Calib/OrbitReset/1550600800000
[42159:internal-dpl-ccdb-backend]: [15:44:59][ERROR] Exception caught: Fatal error 
[42159:internal-dpl-ccdb-backend]: [15:45:00][FATAL] Unhandled o2::framework::runtime_error reached the top of main of o2-sim-digitizer-workflow, device shutting down. Reason: Fatal error
[ERROR] pid 42159 (internal-dpl-ccdb-backend) crashed with or was killed with exit code 1


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/876325b9a525bf13d82133b23bf00d093e0545e8/slc8_x86-64/o2checkcode/1.0-local313/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12368/e09eedbbb4e1be7a0cbdf573ca61e531800c5646/build_O2_fullCI/pretty.html).

--- a/DataFormats/Detectors/CTP/src/Scalers.cxx
+++ b/DataFormats/Detectors/CTP/src/Scalers.cxx
@@ -641,7 +641,7 @@ std::pair<double, double> CTPRunScalers::getRate(uint32_t orbit, int classindex,
   // orbit numbers
 
   // then we can use binary search to find the right entries
-  auto iter = std::lower_bound(mScalerRecordO2.begin(), mScalerRecordO2.end(), orbit, [&](CTPScalerRecordO2 const& a, uint32_t value) { return a.intRecord.orbit < value; });
+  auto iter = std::lower_bound(mScalerRecordO2.begin(), mScalerRecordO2.end(), orbit, [&](CTPScalerRecordO2 const& a, uint32_t value) { return a.intRecord.orbit <= value; });
   auto nextindex = iter - mScalerRecordO2.begin(); // this points to the first index that has orbit greater or equal to given orbit
 
   auto calcRate = [&](auto index1, auto index2) -> double {
@@ -702,8 +702,11 @@ std::pair<double, double> CTPRunScalers::getRateGivenT(double timestamp, int cla
   // orbit numbers
 
   // then we can use binary search to find the right entries
-  auto iter = std::lower_bound(mScalerRecordO2.begin(), mScalerRecordO2.end(), timestamp, [&](CTPScalerRecordO2 const& a, uint32_t value) { return a.epochTime < value; });
-  auto nextindex = iter - mScalerRecordO2.begin(); // this points to the first index that has orbit greater or equal to given orbit
+  auto iter = std::lower_bound(mScalerRecordO2.begin(), mScalerRecordO2.end(), timestamp, [&](CTPScalerRecordO2 const& a, double value) { return a.epochTime <= value; });
+  // this points to the first index that has orbit greater to given orbit;
+  // If this is 0, it means that the above condition was false from the beginning, basically saying that the timestamp is below any of the ScalerRecords' orbits.
+  // If this is mScalerRecordO2.size(), it means mScalerRecordO2.end() was returned, condition was met throughout all ScalerRecords, basically saying the timestamp is above any of the ScalarRecordss orbits.
+  auto nextindex = iter - mScalerRecordO2.begin();
 
   auto calcRate = [&](auto index1, auto index2) -> double {
     auto next = &mScalerRecordO2[index2];

